app/src/main/java/fr/free/nrw/commons/depictions/Media/DepictedImagesFragment.java

```diff
@@ -0,0 +1,264 @@
+     * Display snackbar
+     */
+    @Override
+    public void showSnackBar() {
+        ViewUtil.showShortSnackbar(parentLayout, R.string.error_loading_images);
+    }
+
+    /**
+     * Set visibility of progressbar depending on the boolean value
+     */
+    @Override
+    public void progressBarVisible(Boolean value) {
+        if (value) {
+            progressBar.setVisibility(VISIBLE);
+        } else {
+            progressBar.setVisibility(GONE);
+        }
+    }
+
+    /**
+     * It return an instance of gridView adapter which helps in extracting media details
+     * used by the gridView
+     *
+     * @return GridView Adapter
+     */
+    @Override
+    public ListAdapter getAdapter() {
+        return gridAdapter;
+    }
+
+    /**
+     * Initializes the adapter with a list of Media objects
+     *
+     * @param mediaList List of new Media to be displayed
+     */
+    @Override
+    public void setAdapter(List<Media> mediaList) {
+        gridAdapter = new fr.free.nrw.commons.depictions.GridViewAdapter(getContext(), R.layout.layout_depict_image, mediaList);
+        gridView.setAdapter(gridAdapter);
+    }
+
+    /**
+     * adds list to adapter
+     */
+    @Override
+    public void addItemsToAdapter(List<Media> media) {
+        gridAdapter.addAll(media);
+        gridAdapter.notifyDataSetChanged();
+    }
+
+    /**
+     * Sets loading status depending on the boolean value
+     */
+    @Override
+    public void setLoadingStatus(Boolean value) {
+        if (!value) {
+            statusTextView.setVisibility(GONE);
+        }
+        isLoading = value;
+    }
+
+    /**
+     * Inform the view that there are no more items to be loaded for this search query
+     * or reset the isLastPage for the current query
+     * @param isLastPage
+     */
+    @Override
+    public void setIsLastPage(boolean isLastPage) {
+        this.isLastPage=isLastPage;
+        progressBar.setVisibility(GONE);
+    }
+
+
+    /**
+     * Handles the success scenario
+     * On first load, it initializes the grid view. On subsequent loads, it adds items to the adapter
+     *
+     * @param collection List of new Media to be displayed
+     */
+    @Override
+    public void handleSuccess(List<Media> collection) {
+       presenter.addItemsToQueryList(collection);
+        if (gridAdapter == null) {
+            setAdapter(collection);
+        } else {
+            if (gridAdapter.containsAll(collection)) {
+                return;
+            }
+            gridAdapter.addItems(collection);
+
+            try {
+                ((WikidataItemDetailsActivity) getContext()).viewPagerNotifyDataSetChanged();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        progressBar.setVisibility(GONE);
+        isLoading = false;
+        statusTextView.setVisibility(GONE);
+        for (Media m : collection) {
+            presenter.replaceTitlesWithCaptions("M"+m.getPageId(), mediaSize++);
```

Comment:
The prefix `M` is used at several places. It could be extracted out as a constant.